runtime/CSharp/src/Atn/ParserATNSimulator.cs

```diff
@@ -1399,7 +1397,7 @@ protected int getAltThatFinishedDecisionEntryRule(ATNConfigSet configSet)
 		 *  Assumption: the input stream has been restored to the starting point
 		 *  prediction, which is where predicates need to evaluate.
 		 */
-		protected Pair<ATNConfigSet, ATNConfigSet> SplitAccordingToSemanticValidity(
+		protected KeyValuePair<ATNConfigSet, ATNConfigSet> SplitAccordingToSemanticValidity(
```

Comment:
Not sure I'm supportive. Although KVP is technically equivalent, it is semantically inappropriate. Pair is simple enough that we can keep it.

Reply:
Ok, [`Tuple`](https://docs.microsoft.com/en-us/dotnet/api/system.tuple-2?view=net-6.0) is equivalent in this case. But it's `class` instead of `struct` it may be less effective.

Reply:
Tuple seems appropriate

Reply:
true. KVP might be misleading.